models/demos/t3000/falcon40b/tt/falcon_attention.py

```diff
@@ -249,6 +250,49 @@ def __init__(
         # self.scalar = pad_by_zero(torch.Tensor([1 / math.sqrt(self.head_dim)]), self.device)[0]
         self.scalar = 1 / math.sqrt(self.head_dim)
 
+        # Preloading the kvcache
+        attn_cache_shape = (
+            self.max_batch_size,
+            self.num_kv_heads // len(devices),
+            self.max_position_embeddings,
+            # 2048 + 128,  # Meets benchmarking spec needs
+            self.head_dim,
+        )
+        kvcache_path = tt_cache_path / f"empty_attn_cache{attn_cache_shape}.bin"
+        k_cache = []
+        v_cache = []
+        if (kvcache_path).exists():
+            for i in range(len(self.devices)):
+                k_cache.append(
+                    tt_lib.tensor.load_tensor(str(kvcache_path)).to(devices[i], self.model_config["DRAM_MEMCFG"])
+                )
+                v_cache.append(
+                    tt_lib.tensor.load_tensor(str(kvcache_path)).to(devices[i], self.model_config["DRAM_MEMCFG"])
+                )
```

Comment:
Might save some overhead if load_tensor is called once since cache is tilized on host right?

Reply:
The tensor is tilized before saving it to disk, so I think this call only loads the tensor from disk and directly pushed it to the device without any conversion.